include/jsonserializer.h

```diff
@@ -0,0 +1,102 @@
+//+--------------------------------------------------------------------------
+//
+// File:        jsonserializer.h
+//
+// NightDriverStrip - (c) 2018 Plummer's Software LLC.  All Rights Reserved.  
+//
+// This file is part of the NightDriver software project.
+//
+//    NightDriver is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//   
+//    NightDriver is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU General Public License for more details.
+//   
+//    You should have received a copy of the GNU General Public License
+//    along with Nightdriver.  It is normally found in copying.txt
+//    If not, see <https://www.gnu.org/licenses/>.
+//
+// Description:
+//
+//    Declares classes for JSON (de)serialization of selected 
+//    properties of selected classes
+//
+// History:     Mar-29-2023         Rbergen      Created for NightDriverStrip
+//              
+//---------------------------------------------------------------------------
+
+#pragma once
+
+#include <ArduinoJson.h>
+#include "jsonbase.h"
+#include "FastLED.h"
+
+struct IJSONSerializable

```

Comment:
Using a struct means the functions will be public, just confirming that's what you want here!

Reply:
Yes, it is. This is the "serializable" interface definition that declares one method, which must be public to be useful. As it is an interface definition, I expect any future additions to have to be public too.